Dynamics for multis

miroslavbroz opened the pull request:
Re-implementation of Keperian dynamics for N bodies, so that it is compatible with Rebound numerical integrator.

miroslavbroz committed:
A version working for triple systems!

miroslavbroz committed:
Triple systems with two if's.

miroslavbroz committed:
Correction of nbody.py (pip3 install rebound).
Note: Remeshing does not work. :-(

miroslavbroz committed:
Correction of the Euler angle.

miroslavbroz committed:
Correction of constraints <- double precision!

miroslavbroz committed:
Epsilon parameter for integrators.

miroslavbroz committed:
A re-implementation of keplerian.py for triples!
Elements in Jacobi coordinates -> barycentric frame.
A 1:1 correspondence to nbody.py (rebound).
As of now, no ltte, Euler angles, dpdt, dperdt, ...

miroslavbroz committed:
Correction of constraints <- "%.17e" format.

miroslavbroz committed:
solRad units in Keplerian dynamics.

miroslavbroz committed:
Computation of the Euler angles.

miroslavbroz committed:
vgamma in Keplerian dynamics.

miroslavbroz committed:
Correction of vgamma unit.

miroslavbroz committed:
dpdt, dperdt in Keplerian dynamics.

miroslavbroz committed:
Documentation of Keplerian dynamics.

miroslavbroz committed:
ltte in Keplerian dynamics.

miroslavbroz committed:
Rebuilt default binaries.

miroslavbroz committed:
ASCII art.

miroslavbroz committed:
Introducing a "xyz" dynamics with a geometry layer.

miroslavbroz committed:
Pass roche **kwarg parameters.

miroslavbroz committed:
Don't reuse module variable (cape).

miroslavbroz committed:
The "xyz" dynamics with an inverse geometry layer.
Euler angles.
Roche parameters.
Hierarchical systems.
Twopairs systems.

miroslavbroz committed:
Correction of orientation and Euler angles.

miroslavbroz committed:
Reboundx support.
If import reboundx doesn't work, use workaround
export LD_LIBRARY_PATH=~/.local/lib/python3.9/site-packages/

miroslavbroz committed:
Reboundx update.

miroslavbroz committed:
Correction of Euler angles for triples.

miroslavbroz committed:
Single star without ds, Fs.

miroslavbroz committed:
The "xyz" dynamics documentation.

miroslavbroz committed:
J2 = -C20 or oblateness.

miroslavbroz committed:
Rebuilt default binaries.

miroslavbroz committed:
Merge branch 'feature-multis' into feature-multis

miroslavbroz committed:
Correction of J2 (requivs).

miroslavbroz committed:
Merge branch 'feature-multis' of github.com:miroslavbroz/phoebe2 into feature-multis

aprsa commented on phoebe/dynamics/coord_b2j.py:
```diff
@@ -0,0 +1,60 @@
+#!/usr/bin/env python3
```
Can you please consolidate all source files related to transformations into a single python file, say transformations.py, so that we have everything in one place? Also, could you move executable parts to a test suite so that it's run alongside other tests? Finally, could you write up a high level documentation for the new implementation? It would be good to put our heads together and discuss naming -- for example, 'xyz' vs. 'cartesian' and similar.

miroslavbroz committed:
Use 'gr_full' in reboundx.

miroslavbroz committed:
Correction of J2 = -C20 in reboundx.
Using user-defined force 'j2' instead of 'gravitational harmonics',
where the z-axis was assumed to be fixed,
not perpendicular to the respective orbit(s).
For the moment, 'j2' must be added as follows:

https://sirrah.troja.mff.cuni.cz/~mira/tmp/phoebe2/reboundx/

miroslavbroz committed:
Consolidated to coord.py, orbel.py, geometry.py.

miroslavbroz committed:
Correction of import.

miroslavbroz committed:
Test of geometry.

miroslavbroz committed:
Generalisation of J2 = -C20 in reboundx.
Each component has its own rotation axis,
with respect to which the acceleration must be computed
(and transformed back to the inertial frame).
For the moment, src/gravitational_harmonics.c must be modified:

https://sirrah.troja.mff.cuni.cz/~mira/tmp/phoebe2/reboundx_j2_/

miroslavbroz committed:
Round-offs in elements.

miroslavbroz committed:
Mass constraint for multiples.

miroslavbroz committed:
Quadratic term in the mean anomaly.

miroslavbroz committed:
Corrected epoch (t0) for N-body dynamics.
